### DISCUSSION TITLE
Icepak fluid region mesh

### DISCUSSION THREAD

**Discussion by ghost** [ansys/pyaedt#4778] [Q&A] [answered]
I am in process of running a board level simulation using pyaedt for Icepak, I am facing issue with meshing , details of which can be found below

1. For meshing when manual method is turned off (icepak.mesh.global_mesh_region.manual_settings= False) I can see mesh both in materials as well as region created(fluid domain)

2. When manual setting (icepak.mesh.global_mesh_region.manual_settings= True ) is turned on and maximum element sizes are given using icepak.mesh.global_mesh_region.settings["MaxElementSizeX"]="Xmm" and then updated using icepak.mesh.global_mesh_region.update() command to generate mesh , I can see mesh only in solid domain and not on air domain.

Script used for meshing can be found here
 
 icepak.mesh.global_mesh_region.manual_settings= True
icepak.mesh.global_mesh_region.settings["MaxElementSizeX"]="3mm"
icepak.mesh.global_mesh_region.settings["MaxElementSizeY"]="3mm"
icepak.mesh.global_mesh_region.settings["MaxElementSizeZ"]="0.5mm"
icepak.mesh.global_mesh_region.Level = 2
icepak.mesh.global_mesh_region.update()
icepak.mesh.generate_mesh()

I am wondering whether some additional lines should be added to mesh airdomain or am i making mistake somewhere else.

**Comment by lorenzovecchietti** [comment, marked as answer]
Hi @BhagirathaSomayaji, thanks for pointing out this behavior. I was able to reproduce this and I will investigate (#4788).
In the meantime, you can use this as a workaround:
```
import pyaedt
icepak = pyaedt.Icepak()
icepak.modeler.create_box([0,0,0], [10,20,30])
icepak.mesh.global_mesh_region.manual_settings= True
icepak.mesh.global_mesh_region.settings["MaxElementSizeX"]="3mm"
icepak.mesh.global_mesh_region.settings["MaxElementSizeY"]="3mm"
icepak.mesh.global_mesh_region.settings["MaxElementSizeZ"]="0.5mm"
icepak.mesh.global_mesh_region.update()
icepak.mesh.global_mesh_region.global_region.object.solve_inside = False
icepak.mesh.global_mesh_region.global_region.object.solve_inside = True
icepak.mesh.generate_mesh()
icepak.close_desktop()
```

`icepak.mesh.global_mesh_region.Level = 2` should not be used for manual mesh settings (#4789).

**Reply by ghost** [reply]
I tried above method, though I am able to create a box , mesh is not getting generated.  

Commands i used are as below
icepak.modeler.create_box([-100,-50,-50], [120,120,120])
icepak.mesh.global_mesh_region.manual_settings= True
icepak.mesh.global_mesh_region.settings["MaxElementSizeX"]="1mm"
Icepak.mesh.global_mesh_region.settings["MaxElementSizeY"]="1mm"
icepak.mesh.global_mesh_region.settings["MaxElementSizeZ"]="0.5mm"
icepak.mesh.global_mesh_region.update()
icepak.mesh.global_mesh_region.global_region.object.solve_inside = False
icepak.mesh.global_mesh_region.global_region.object.solve_inside = True

I have recieved error message which is 

PyAEDT WARNING: No mesh operation found.
PyAEDT WARNING: No mesh region found.
PyAEDT INFO: Mesh class has been initialized! Elapsed time: 0m 0sec
Traceback (most recent call last):

  File C:\ProgramData\anaconda3\Lib\site-packages\spyder_kernels\py3compat.py:356 in compat_exec
    exec(code, globals, locals)

  File c:\aedt_sripting\scripting_example.py:79
    Icepak.mesh.global_mesh_region.settings["MaxElementSizeY"]="1mm"

AttributeError: 'function' object has no attribute 'mesh'

Do let me know your suggestion on this.

**Reply by lorenzovecchietti** [reply]
Hi, I just tried again and everything is working smoothly on my side. Please try to update pyaedt to the latest version.

If you still have any problem with the updated version, please provide the python and the AEDT version you are using.

**Reply by ghost** [reply]
Hi, I am using Ansys 2023R2 and 0.9.4

**Reply by lorenzovecchietti** [reply]
Can you try 0.9.5 that was just released? In the meantime, I'll test on 2023R2.
